### ISSUE TITLE
Error when listing Amazon Prime games

### ISSUE THREAD

**Issue by lelecanfora** [lutris/lutris#5702]
### Bug description

After successful authentication, I get an error when trying to retrieve the games list.
```
'NoneType' object is not iterable

Traceback (most recent call last):

  File "/usr/lib/python3/dist-packages/lutris/util/jobs.py", line 30, in target
    result = self.function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3/dist-packages/lutris/services/base.py", line 142, in do_reload
    self.load()

  File "/usr/lib/python3/dist-packages/lutris/services/amazon.py", line 160, in load
    games = [AmazonGame.new_from_amazon_game(game) for game in self.get_library()]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

TypeError: 'NoneType' object is not iterable
```

### How to Reproduce

1. Log in to Amazon Prime games on Lutris
2. Click on reload list

### Expected behavior

The games list loads successfully

### Log output

```shell
2024-11-04 14:20:39,080: The Battle.net source is unavailable because Google protobuf could not be loaded: No module named 'google'
INFO     2024-11-04 14:20:39,268 [application.do_command_line:479]:Starting Lutris 0.5.17
INFO     2024-11-04 14:20:39,325 [startup.run_all_checks:120]:AMD Radeon Graphics (1002:1638 17aa:5095 amdgpu) Driver 24.0.9
DEBUG    2024-11-04 14:20:39,578 [path_cache._update_missing_games:114]:Checking for missing games
DEBUG    2024-11-04 14:20:39,623 [api.check_stale_runtime_versions:52]:Modified at Mo 04 Nov 2024 14:16:46 , will update after Mo 04 Nov 2024 20:16:46 
DEBUG    2024-11-04 14:20:39,639 [lutriswindow.create_runtime_updater_cb:1244]:Runtime up to date
DEBUG    2024-11-04 14:20:44,163 [http._request:88]:GET https://api.amazon.com/user/profile
DEBUG    2024-11-04 14:20:44,637 [base.wipe_game_cache:441]:Deleting amazon cache /home/lele/.cache/lutris/amazon-library.json
DEBUG    2024-11-04 14:20:44,637 [base.wipe_game_cache:175]:Deleting games from service-games for amazon
DEBUG    2024-11-04 14:20:44,639 [http._request:88]:GET https://api.amazon.com/user/profile
DEBUG    2024-11-04 14:20:45,106 [http._request:88]:POST https://sds.amazon.com/amazon/
ERROR    2024-11-04 14:20:45,514 [amazon.request_sds:417]:Failed http request https://sds.amazon.com/amazon/: HTTP Error 503: Service Temporarily Unavailable
ERROR    2024-11-04 14:20:45,515 [jobs.target:32]:Error while completing task <function BaseService.start_reload.<locals>.do_reload at 0x7d20ad46ede0>: <class 'TypeError'> 'NoneType' object is not iterable
  File "/usr/lib/python3/dist-packages/lutris/util/jobs.py", line 30, in target
    result = self.function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/lutris/services/base.py", line 142, in do_reload
    self.load()
  File "/usr/lib/python3/dist-packages/lutris/services/amazon.py", line 160, in load
    games = [AmazonGame.new_from_amazon_game(game) for game in self.get_library()]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INFO     2024-11-04 14:21:07,434 [application.do_shutdown:1051]:Shutting down Lutris
```

### System Information

```shell
Vulkan support: YES
Esync support: YES
Fsync support: YES
Wine installed: YES
Gamescope: YES
Mangohud: YES
Gamemode: YES
Steam: YES
In Flatpak: NO
[System]
OS: Ubuntu 24.04 Noble Numbat
Arch: x86_64
Kernel: 6.8.0-48-generic
Desktop: ubuntu:GNOME
Display Server: wayland
[CPU]
Vendor: AuthenticAMD
Model: AMD Ryzen 7 PRO 5850U with Radeon Graphics
Physical cores: 8
Logical cores: 16
[Memory]
RAM: 27.2 GB
Swap: 8.0 GB
[Graphics]
Vendor: AMD
OpenGL Renderer: AMD Radeon Graphics (radeonsi, renoir, LLVM 17.0.6, DRM 3.57, 6.8.0-48-generic)
OpenGL Version: 4.6 (Compatibility Profile) Mesa 24.0.9-0ubuntu0.2
OpenGL Core: 4.6 (Core Profile) Mesa 24.0.9-0ubuntu0.2
OpenGL ES: OpenGL ES 3.2 Mesa 24.0.9-0ubuntu0.2
Vulkan Version: 1.3.275
Vulkan Drivers: AMD Radeon Graphics (RADV RENOIR) (1.3.274)
```

**Comment by Katataf1sh** [comment]
I'm getting the exact same error.

**Comment by danieljohnson2** [comment]
This is a duplicate of #5596.

You may wish to [download the master branch](https://github.com/lutris/lutris) and run the ``bin/lutris`` script within; no need to install. I believe the master branch will be able to reload your Amazon source.

This set of fixes should be in release 0.5.18, real soon now!

**Comment by Katataf1sh** [comment]
> This is a duplicate of #5596.
> 
> You may wish to [download the master branch](https://github.com/lutris/lutris) and run the `bin/lutris` script within; no need to install. I believe the master branch will be able to reload your Amazon source.
> 
> This set of fixes should be in release 0.5.18, real soon now!

Just wanted to confirm that downloading the master branch and running the script within bin/lutris indeed fixed the sync problem with Amazon Prime Gaming.
